### PR TITLE
Ports /tg/ advanced surgerytools and tool defines.

### PR DESCRIPTION
**Pull request by lolman360**
## About The Pull Request
Ports https://github.com/tgstation/tgstation/pull/46856.
surgery tools now work on tool defines
chaplain's chainsaw weapons can now do surgery slowly (the admin only inquisition chainsaw sword can do surgery FASTLY)
the mecha drill, chainsaw and mounted chainsaw and CLAMP can now do surgery with 100% accuracy to "promote fun mechanics" haha

## Why It's Good For The Game

makes advanced surgery tools actually good and makes some fun mechanics reward you with high surgery chances

## Changelog
:cl:
tweak: chainsaw kind of weapons and the mecha drill and the CLAMP can now be used with 100% accuracy for surgery
refactor: surgery tools now work on defines
balance: advanced surgerytools can now switch types instead of  just being faster
/:cl:

### CODE REVIEW

File: code/modules/research/designs/medical_designs.dm [hunk 1]
```diff
@@ -590,63 +590,34 @@
 
 /datum/design/retractor_adv
 	name = "Advanced Retractor"
-	desc = "A high-class, premium retractor, featuring precision crafted, silver-plated hook-ends and an electrum handle."
+	desc = "An almagation of rods and gears, able to function as both a surgical clamp and retractor. "
 	id = "retractor_adv"
 	build_type = PROTOLATHE
 	materials = list(MAT_METAL = 500, MAT_GLASS = 500, MAT_SILVER = 1500, MAT_GOLD = 1000)
-	build_path = /obj/item/retractor/adv
```
Comment (Trilbyspaceclone) [inline code/modules/research/designs/medical_designs.dm:597]: Why are you removing adv surgery tools?

File: code/modules/research/techweb/all_nodes.dm [hunk 2]
```diff
@@ -96,7 +96,7 @@
 	display_name = "Advanced Surgery Tools"
 	description = "Refined and improved redesigns for the run-of-the-mill medical utensils."
 	prereq_ids = list("adv_biotech", "adv_surgery")
-	design_ids = list("drapes", "retractor_adv", "hemostat_adv", "cautery_adv", "surgicaldrill_adv", "scalpel_adv", "circular_saw_adv")
```
Comment (Trilbyspaceclone) [inline code/modules/research/techweb/all_nodes.dm:99]: Why are you removing the adv surgery tools?

File: tgstation.dme [hunk 3]
```diff
@@ -517,7 +517,6 @@
 #include "code\game\atoms.dm"
 #include "code\game\atoms_movable.dm"
 #include "code\game\communications.dm"
-#include "code\game\data_huds.dm"
```
Comment (Trilbyspaceclone) [inline tgstation.dme]: ?

Comment (lolman360) [reply tgstation.dme]: accident

**Review by silicons — APPROVED** [review]

**Pull request merged**